example/tutorial.cpp

```diff
@@ -2,21 +2,45 @@
 #include <voxelized_geometry_tools/collision_map.hpp>
 #include <voxelized_geometry_tools/signed_distance_field.hpp>
 #include <voxelized_geometry_tools/ros_interface.hpp>
+#if VOXELIZED_GEOMETRY_TOOLS__SUPPORTED_ROS_VERSION == 2
+#include <rclcpp/rclcpp.hpp>
+#include <std_msgs/msg/color_rgba.hpp>
+#include <visualization_msgs/msg/marker_array.hpp>
+#elif VOXELIZED_GEOMETRY_TOOLS__SUPPORTED_ROS_VERSION == 1
 #include <ros/ros.h>
+#include <std_msgs/ColorRGBA.h>
 #include <visualization_msgs/MarkerArray.h>
+#else
+#error "Undefined or unknown VOXELIZED_GEOMETRY_TOOLS__SUPPORTED_ROS_VERSION"
+#endif
 #include <functional>
 #include <common_robotics_utilities/conversions.hpp>
 #include <common_robotics_utilities/color_builder.hpp>
 
 int main(int argc, char** argv)
 {
+#if VOXELIZED_GEOMETRY_TOOLS__SUPPORTED_ROS_VERSION == 2
+    using ColorRGBA = std_msgs::msg::ColorRGBA;
+    using Marker = visualization_msgs::msg::Marker;
+
+    rclcpp::init(argc, argv);
+    auto node = std::make_shared<rclcpp::Node>("sdf_tools_tutorial");
+    auto visualization_pub_ptr = node->create_publisher<Marker>(
+        "sdf_tools_tutorial_visualization", rclcpp::QoS(1).transient_local());
+#elif VOXELIZED_GEOMETRY_TOOLS__SUPPORTED_ROS_VERSION == 1
+    using ColorRGBA = std_msgs::ColorRGBA;
+    using Marker = visualization_msgs::Marker;
+
     // Make a ROS node, which we'll use to publish copies of the data in the CollisionMap and SDF
     // and Rviz markers that allow us to visualize them.
     ros::init(argc, argv, "sdf_tools_tutorial");
     // Get a handle to the current node
     ros::NodeHandle nh;
     // Make a publisher for visualization messages
-    ros::Publisher visualization_pub = nh.advertise<visualization_msgs::Marker>("sdf_tools_tutorial_visualization", 1, true);
+    ros::Publisher visualization_pub = nh.advertise<Marker>(
+        "sdf_tools_tutorial_visualization", 1, true);
+    ros::Publisher * visualization_pub_ptr = &visualization_pub;
```

Comment:
I'd rather use the
```
const std::function<void(const MarkerArray&)>& display_fn
      = [&] (const MarkerArray& markers)
{
  display_pub->publish(markers);
};
```
approach here to be consistent with the other examples and encapsulate the difference in publisher types.

Reply:
Sounds reasonable. See https://github.com/calderpg/voxelized_geometry_tools/pull/6/commits/33209c215cd7c3fa40dd9bf26d90387e1919f3e2.